computation/src/main/java/com/powsybl/computation/LazyCreatedComputationManager.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.computation;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class LazyCreatedComputationManager implements ComputationManager {
+
+    private final ComputationManagerFactory factory;
+
+    private ComputationManager delegate;
+
+    public LazyCreatedComputationManager(ComputationManagerFactory factory) {
+        this.factory = Objects.requireNonNull(factory);
+    }
+
+    private synchronized ComputationManager getComputationManager() {
+        if (delegate == null) {
+            delegate = factory.create();
+            if (delegate == null) {
+                throw new NullPointerException("Computation manager factory returned null");
+            }
```

Comment:
Don't really understand sonar error, but a Objects.requireNonNull may be more expressive / readable?